controllers/etcdcluster_controller.go

```diff
@@ -411,6 +412,20 @@ func (r *EtcdClusterReconciler) reconcile(
 
 			for _, peer := range peers.Items {
 				if !memberNames.Has(peer.Name) {
+					if !peer.DeletionTimestamp.IsZero() {
+						log.V(2).Info("Peer is already marked for deletion", "peer", peer.Name)
+						continue
```

Comment:
If an EtcdPeer is marked for deletion, is it safe to 'continue' here and potentially mark a second one for deletion? A cursory thought suggests this _is_ fine, because the members will be removed from the membership API.

Reply:
Alternative would be to wait for each deleted peer to disappear before continuing.
But that could lead to a situation where whilst deleting EtcdPeers after a scale-down, 
a user scales-up again and the EtcdPeer for the removed member now can't join the cluster because its data is stamped with the previous member ID.